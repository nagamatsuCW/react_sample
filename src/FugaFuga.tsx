import * as React from 'react'
import {
  Link,
  useNavigate
} from 'react-router-dom';
const FugaFuga = () => {
  const navigate = useNavigate();
  return(
    <div>
      FugaaaPiyoooooo
    </div>
  )
}

export default FugaFuga