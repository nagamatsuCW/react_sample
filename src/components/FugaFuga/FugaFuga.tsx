import * as React from 'react'
import { Props } from './index'

const FugaFuga:React.FC<Props> = (props) => {
  return(
    <div id="test">
      FugaaaPiyoooooo
      <div id="count">
        {props.count}
      </div>
      <button id="incrementButton" onClick={props.increment}>increment</button>
    </div>
  )
}

export default FugaFuga