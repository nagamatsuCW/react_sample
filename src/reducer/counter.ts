import { ActionsType } from '../action'
export interface CounterState {
  count: number
}

const initialState: CounterState = {
  count: 0
};

const counter = (state:CounterState = initialState, action:ActionsType) => {
  switch (action.type) {
    case 'INCREMENT':

      return {
        ...state,
        count: state.count + 1
      }
    case 'DECREMENT':
      return {
        ...state,
        count: state.count - 1
      }
    default:
      return state
  }
}

export default counter