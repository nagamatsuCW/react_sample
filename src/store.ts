import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import createRootReducer, { StoreState } from './reducer'
import { createBrowserHistory } from 'history'
export const history = createBrowserHistory()

const composeEnhancers = (window && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;
const enhancer = composeEnhancers(
  applyMiddleware(
    routerMiddleware(history)
    // other store enhancers if any
  )
);


export default function configureStore(preloadedState: StoreState) {
  const store = createStore(
    createRootReducer(history),
    preloadedState,
    enhancer
  )

  return store
}