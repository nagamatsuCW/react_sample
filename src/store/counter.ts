import { createSlice } from "@reduxjs/toolkit";

export interface CounterState {
  count: number,
  name: string
}

export const initialState: CounterState = {
  count: 1,
  name: ""
};


// Sliceを生成する
const slice = createSlice({
  name: "counter",
  initialState,
  reducers: {
    increment: state => {
      return Object.assign({}, state, { count: state.count + 1 })
    },
    decrement: state => {
          return Object.assign({}, state, { count: state.count + 1 })
    },
    nameChange: (state, action) => {
          return Object.assign({}, state, { name: action.payload })
    },
    // etc...
  }
});

// Reducerをエクスポートする
export default slice.reducer;

// Action Creatorsをエクスポートする
export const { increment, decrement, nameChange } = slice.actions;

export type ActionsType = ReturnType<typeof increment | typeof decrement | typeof nameChange>