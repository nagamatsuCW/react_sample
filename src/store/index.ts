import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";

import { createStore, applyMiddleware, compose } from 'redux';
import { connectRouter, routerMiddleware } from 'connected-react-router'
import { createBrowserHistory } from 'history'
import logger from 'redux-logger'

// それぞれ slice.reducer を default export している前提
import counterReducer from "./counter";


export const history = createBrowserHistory()

const composeEnhancers = (window && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;
const enhancer = composeEnhancers(
  applyMiddleware(
    routerMiddleware(history),
    logger
  )
);

const reducer = combineReducers({
  router: connectRouter(history),
  counter: counterReducer,
})

export type RootState = ReturnType<typeof reducer>

const store = configureStore({
  reducer,
  middleware(getDefaultMiddleware) {
    return getDefaultMiddleware().concat(routerMiddleware(history))
  },
  devTools: process.env.NODE_ENV !== 'production',
  enhancers: [enhancer],
})

export default store
