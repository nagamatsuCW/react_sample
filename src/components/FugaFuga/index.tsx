import { connect } from "react-redux";
import { increment, nameChange } from '../../action'
import { StoreState } from '../../reducer'
import { Dispatch } from 'redux'
import FugaFuga from './FugaFuga'


function mapStateToProps(state:StoreState) {
  return {
      count: state.counter.count,
      name: state.counter.name
  };
}

function mapDispatchToProps(dispatch: Dispatch) {
  return {
    increment: () => { dispatch(increment()) },
    nameChange: (value: string) => { dispatch(nameChange({name: value})) }
  }
}

// type Props = ownProps & RetuZrnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;
export type Props = ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FugaFuga);
