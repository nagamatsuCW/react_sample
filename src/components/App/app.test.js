import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import FugaFuga from '../FugaFuga/index';
import { increment } from '../../store/counter'
import { useSelector, useDispatch } from 'react-redux'
const useSelectorMock = jest.fn()
const useDispatchMock = jest.fn()

Enzyme.configure({ adapter: new Adapter() });

function sum(a, b) {
  return a + b;
}
module.exports = sum;


// 事前準備
jest.mock('react-redux')

test('adds 1 + 2 to equal 3', () => {
  expect(sum(1, 2)).toBe(3);
});

describe('FugaFuga.tsx', () => {
  afterEach(() => {
    jest.resetAllMocks()
  })
  beforeEach(() => {
    useSelectorMock.mockReturnValue({count: 0})
    useDispatchMock.mockReturnValue(jest.fn())
  })

it('dispatches increment action', () => {
    const wrapper = shallow(<FugaFuga />)
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('#test').text()).toEqual('FugaaaPiyoooooo名前: increment');
  })
})