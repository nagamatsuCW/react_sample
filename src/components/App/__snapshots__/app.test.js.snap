// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FugaFuga.tsx dispatches increment action 1`] = `ShallowWrapper {}`;
